### PR TITLE
fdroidserver: unstable-2023-10-23 -> 2.2.2

### PR DESCRIPTION
**Pull request by natsukium**
Changelog:
https://gitlab.com/fdroid/fdroidserver/-/blob/2.2.2/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/fdroidserver/default.nix [hunk 1]
```diff
@@ -82,13 +83,8 @@ buildPythonApplication rec {
...
     qrcode
     requests
-    (ruamel-yaml.overrideAttrs (old: {
-      src = fetchPypi {
-        pname = "ruamel.yaml";
-        version = "0.17.21";
-        hash = "sha256-i3zml6LyEnUqNcGsQURx3BbEJMlXO+SSa1b/P10jt68=";
-      };
-    }))
+    ruamel-yaml
```
Comment (linsui) [inline pkgs/development/tools/fdroidserver/default.nix:86]: Why do you remove this override? Given that F-Droid buildserver is still running on Debian bookworm which has 0.17.21 we still need to use this version.

Comment (natsukium) [reply pkgs/development/tools/fdroidserver/default.nix:86]: because I couldn't find a reason to fix the version anywhere.

Comment (linsui) [reply pkgs/development/tools/fdroidserver/default.nix:86]: See https://github.com/NixOS/nixpkgs/pull/254518.

File: pkgs/development/tools/fdroidserver/default.nix [hunk 2]
```diff
@@ -22,21 +20,25 @@
...
+, sdkmanager
+, setuptools
 , yamllint
 }:
 
 buildPythonApplication rec {
   pname = "fdroidserver";
-  version = "unstable-2023-10-23";
-  format = "setuptools";
+  version = "2.2.2";
```
Comment (linsui) [inline pkgs/development/tools/fdroidserver/default.nix:30]: This is mostly a downgrade and breaks the format function. Please update to 2.3a1 instead. We shuold also update androguard to 4.1 maybe with an override.

**Comment by natsukium** [comment]
I apologize for taking the time to review this package, but I'll withdraw this PR because I can't properly test it.
I'm not familiar with this package itself, as I only found it while refactoring the python package set.
I think we can save a lot of time if the source code or the commit message describes which version to adopt and which package needs to be fixed, and if the tests for expected behavior are implemented.

**Comment by linsui** [comment]
Can you tell me what the refactor about? I'll add them when I update this package.

**Pull request closed without merging**